paymentsheet/src/test/java/com/stripe/android/paymentsheet/repositories/CustomerAdapterTest.kt

```diff
@@ -26,26 +32,83 @@ class CustomerAdapterTest {
 
     @Test
     fun `CustomerAdapter can be created`() {
-        val customerEphemeralKeyProvider = CustomerEphemeralKeyProvider {
-            Result.success(
-                CustomerEphemeralKey(
-                    customerId = "cus_123",
-                    ephemeralKey = "ek_123"
+        val adapter = createAdapter()
+        assertThat(adapter).isNotNull()
+    }
+
+    @Test
+    fun `CustomerEphemeralKey is cached`() = runTest {
+        val adapter = createAdapter(
+            customerEphemeralKeyProvider = {
+                Result.success(
+                    CustomerEphemeralKey(
+                        customerId = testScheduler.currentTime.toString(),
+                        ephemeralKey = "ek_123"
+                    )
                 )
-            )
-        }
+            },
+            timeProvider = { testScheduler.currentTime }
+        )
+        assertThat(adapter.cachedCustomer).isNull()
 
-        val setupIntentClientSecretProvider = SetupIntentClientSecretProvider {
-            Result.success("seti_123")
-        }
+        var customer = adapter.getCustomer()
+        assertThat(customer.getOrNull()?.customerId).isEqualTo("0")
+        assertThat(adapter.cachedCustomer).isNotNull()
 
-        val adapter = CustomerAdapter.create(
```

Comment:
Looks like we lose our test for this flow (using dagger to create via the factory). We should keep at least one test using this old method.